stdlib/stdlib.mli

```diff
@@ -553,6 +574,66 @@ external tanh : float -> float = "caml_tanh_float" "tanh"
   [@@unboxed] [@@noalloc]
 (** Hyperbolic tangent.  Argument is in radians. *)
 
+external acosh : float -> float = "caml_acosh_float" "caml_acosh"
+  [@@unboxed] [@@noalloc]
+(** Hyperbolic arc cosine.  The argument must fall within the range
+    [[1.0, inf]].
+    Result is in radians and is between [0.0] and [inf].
+
+    @since 4.10.0
+*)
+
+external asinh : float -> float = "caml_asinh_float" "caml_asinh"
+  [@@unboxed] [@@noalloc]
+(** Hyperbolic arc sine.  The argument and result range over the entire
+    real line.
+    Result is in radians.
+
+    @since 4.10.0
+*)
+
+external atanh : float -> float = "caml_atanh_float" "caml_atanh"
+  [@@unboxed] [@@noalloc]
+(** Hyperbolic arc tangent.  The argument must fall within the range
+    [[-1.0, 1.0]].
+    Result is in radians and ranges over the entire real line.
+
+    @since 4.10.0
+*)
+
+external erf : float -> float = "caml_erf_float" "caml_erf"
+  [@@unboxed] [@@noalloc]
+(** Error function.  The argument ranges over the entire real line.
+    The result is always within [[-1.0, 1.0]].
+
+    @since 4.10.0
+*)
+
+external erfc : float -> float = "caml_erfc_float" "caml_erfc"
+  [@@unboxed] [@@noalloc]
+(** Complementary error function ([erfc x = 1.0 -. erf x]).
+    The argument ranges over the entire real line.
+    The result is always within [[-1.0, 1.0]].
+
+    @since 4.10.0
+*)
+
+external trunc : float -> float = "caml_trunc_float" "caml_trunc"
+                                    [@@unboxed] [@@noalloc]
+(** [trunc x] rounds [x] to the nearest integer whose absolute value is
+   less than or equal to [x].
+
+   @since 4.08.0 *)
+
+external round : float -> float = "caml_round_float" "caml_round"
+                                    [@@unboxed] [@@noalloc]
+(** [round x] rounds [x] to the nearest integer with ties (fractional
+   values of 0.5) rounded away from zero, regardless of the current
+   rounding direction.  If [x] is an integer, [+0.], [-0.], [nan], or
+   infinite, [x] itself is returned.
+
+   @since 4.08.0 *)
```

Comment:
Shouldn't these functions only be added to the new `Float` module? This PR adds them to both `Float` and `Stdlib`.

Reply:
I think there's a consensus to no longer add directly to `Stdlib`. All recent floating point operations additions (#1354, #2177, #1794) only went to the `Float` module.

Reply:
I can certainly remove the `Stdlib` functions that were previously in `Float` only (`trunc`, and `round`) and some of the new ones.  But maybe some of the new functions should stay.  E.g. we have `sin`, `cos`, `tan` as well as their inverses `asin`, etc., in `Stdlib`.  We also already have `sinh`, `cosh`, and `tanh` there.   It would seem weirdly inconsistent to not have the hyperbolic inverses (`asinh`, etc.) in `Stdlib`.

Considering the feedback, my suggestion would be to remove all additions to `Stdlib` from this patch except for `asinh`, `acosh`, and `atanh`, which should remain for consistency.

Reply:
I'm not the one who has the power on this. But my opinion is that if people want to be consistent they should always use the `Float` module and forget about the  `Stdlib` entry points. Besides these `Stdlib` entry points should be deprecated once the `Float` module hits debian stable, despite the fact that some people want to make us believe that deprecating stuff breaks code (secret: it does not). Then in n more years we might be able to delete them altogheter.

Reply:
I'm generally in favor of eventually moving all float operations to `Float`.  But given that this will likely take a very long time, I don't see the harm in having a consistent set of float operations in `Stdlib` until then.  It's almost certainly not going to make programs any more dependent on deprecated functions than they would be otherwise.  Anyway, I personally don't have strong feelings about this, since I'll switch to using `Float` only in either case.

Reply:
Personally I would go with @mmottl's proposal to remove everything from stdlib except the reverse hyperbolic functions. It sounds like a good compromise to me.